File: app/src/main/java/es/osw/cafelatte/presentation/model/mapper/LoginUserMapper.java
package es.osw.cafelatte.presentation.model.mapper;

import com.firebase.ui.auth.IdpResponse;
import es.osw.cafelatte.domain.model.LoginUser;

public class LoginUserMapper {
  public static LoginUser map(IdpResponse idpResponse) {
    return LoginUser.newBuilder()
        .setEmail(idpResponse.getEmail())
        .setToken(idpResponse.getIdpToken())
        .setProvider(idpResponse.getProviderType())
        .setSecret(idpResponse.getIdpSecret())
        .build();
  }
}

File: app/src/main/java/es/osw/cafelatte/presentation/model/mapper/LoginUserMapper.kt
package es.osw.cafelatte.presentation.model.mapper

import com.firebase.ui.auth.IdpResponse
import es.osw.cafelatte.domain.model.LoginUser

class LoginUserMapper {
    companion object {
        fun map(idpResponse: IdpResponse): LoginUser {
            return LoginUser(
                    email = idpResponse.email,
                    token = idpResponse.idpToken,
                    provider = idpResponse.providerType,
                    secret = idpResponse.idpSecret)
        }
    }

}
